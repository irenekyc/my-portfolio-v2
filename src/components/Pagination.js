import React from 'react'
import {useDispatch, useSelector} from 'react-redux'
import {fetchData} from '../actions/data'


const Pagination = ()=>{
    const dispatch = useDispatch()
    const {curPage, sort, filter, totalPages} = useSelector(state=> state.status)
    return(
        <div className="pagination-bar">
            {curPage>0 &&(<div class="prev">
                <span onClick={()=>dispatch(fetchData(curPage-1, sort, filter))}>Prev </span>
            </div>)}
            <div class="next">
            {curPage < totalPages-1 &&( <span onClick = {()=>dispatch(fetchData(curPage+1, sort, filter))}> Next</span>)}
              
            </div>
        </div>

    )
}

export default Pagination