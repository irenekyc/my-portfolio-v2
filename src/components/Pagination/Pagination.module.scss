.ikPaginationBar {
  width: 80%;
  margin-right: auto;
  margin-left: auto;
  max-width: 900px;
  padding: 2rem 0;
  display: flex;

  &__button {
    font-size: 1rem;
    color: #00adb5;
    cursor: pointer;
    border: none;
    background: none;
    outline: none;

    &:hover {
      color: #eee;
    }

    &__prev {
      margin-left: 0;
      margin-right: auto;
    }
    &__next {
      margin-right: 0;
      margin-left: auto;
    }
  }
}
