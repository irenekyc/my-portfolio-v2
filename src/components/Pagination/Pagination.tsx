import React, { FunctionComponent } from "react";
import styles from "./Pagination.module.scss";
import classnames from "classnames";

interface PagingationProps {
  goToNextPage: () => void;
  goToPrevPage: () => void;
  hasNext: boolean;
  hasPrev: boolean;
}

const Pagination: FunctionComponent<PagingationProps> = ({
  goToNextPage,
  goToPrevPage,
  hasNext,
  hasPrev,
}: PagingationProps) => {
  return (
    <div className={styles.ikPaginationBar} data-testid="pagination">
      {hasPrev && (
        <button
          onClick={goToPrevPage}
          className={classnames(
            styles.ikPaginationBar__button,
            styles.ikPaginationBar__button__prev
          )}
          data-testid="pagination-prev"
        >
          Prev
        </button>
      )}
      {hasNext && (
        <button
          onClick={goToNextPage}
          className={classnames(
            styles.ikPaginationBar__button,
            styles.ikPaginationBar__button__next
          )}
          data-testid="pagination-next"
        >
          Next
        </button>
      )}
    </div>
  );
};

export default Pagination;
