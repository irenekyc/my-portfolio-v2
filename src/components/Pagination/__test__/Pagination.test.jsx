import React from "react";
import Pagination from "../index";
import { render, screen, fireEvent } from "@testing-library/react";
const mockGoToNextPage = jest.fn();
const mockGoToPrevPage = jest.fn();

it("Render Pagination Section - no button", () => {
  render(
    <Pagination
      hasNext={false}
      hasPrev={false}
      goToNextPage={mockGoToNextPage}
      goToPrevPage={mockGoToPrevPage}
    />
  );

  const paginationBar = screen.getByTestId("pagination");
  expect(paginationBar).toBeInTheDocument();
  const prevButton = screen.queryByTestId("pagination-prev");
  const nextButton = screen.queryByTestId("pagination-next");
  expect(prevButton).toBeNull();
  expect(nextButton).toBeNull();
});

it("Render Pagination Section - click on prev and next button", () => {
  render(
    <Pagination
      hasNext={true}
      hasPrev={true}
      goToNextPage={mockGoToNextPage}
      goToPrevPage={mockGoToPrevPage}
    />
  );

  const paginationBar = screen.getByTestId("pagination");
  expect(paginationBar).toBeInTheDocument();
  const prevButton = screen.queryByTestId("pagination-prev");
  const nextButton = screen.queryByTestId("pagination-next");
  expect(prevButton).not.toBeNull();
  expect(nextButton).not.toBeNull();
  fireEvent.click(prevButton);
  expect(mockGoToPrevPage).toBeCalled();
  fireEvent.click(nextButton);
  expect(mockGoToNextPage).toBeCalled();
});
