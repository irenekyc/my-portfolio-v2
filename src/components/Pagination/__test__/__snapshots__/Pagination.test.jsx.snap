// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Pagination Section - click on prev and next button 1`] = `
<div
  class="ikPaginationBar"
  data-testid="pagination"
>
  <button
    class="ikPaginationBar__button ikPaginationBar__button__prev"
    data-testid="pagination-prev"
  >
    Prev
  </button>
  <button
    class="ikPaginationBar__button ikPaginationBar__button__next"
    data-testid="pagination-next"
  >
    Next
  </button>
</div>
`;
